### ISSUE TITLE
Reduce unused JavaScript:

### ISSUE THREAD

**Issue by ormanlis** [vercel/next.js#40829]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

https://nextjs.org/learn/seo/improve/lighthouse

browser chrome

### Describe the Bug

Here mentions that next.js is getting very good scores
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/103314412/191908007-6a3827b2-0246-4ddd-ae53-42d88fed7ba7.png">

but once I run the test I'm getting this

<img width="1308" alt="image" src="https://user-images.githubusercontent.com/103314412/191908040-4a6cf681-979e-45c6-b1da-ee2575974dfa.png">

run test: https://pagespeed.web.dev/report?url=https%3A%2F%2Fnextjs.org%2Flearn%2Fseo%2Fimprove%2Flighthouse

----

There is a huge discrepancy and I'm getting similar problems in my own app. I was wondering what we can do to have something above 90 score?

I think the main problem is that after you load the page, it needs to download and execute the whole js code. I wonder if it's possible to further reduce this initial js code and avoid re-render the page (this could work for static pages).

### Expected Behavior

I expected to be faster, but not sure what is the solution.

### Link to reproduction

https://nextjs.org/learn/seo/improve/lighthouse

### To Reproduce

Go to: https://pagespeed.web.dev/report?url=https%3A%2F%2Fnextjs.org%2Flearn%2Fseo%2Fimprove%2Flighthouse

**Comment by dcyriller** [comment]
@AlexParton yes it is online. You could take a look to the pagespeed report generated for the next.js homepage. (link: https://pagespeed.web.dev/report?url=https%3A%2F%2Fnextjs.org%2F). I see a performance score of 59 on mobile (close to @ormanlis' score).

The first screenshot is generated with Lighthouse embedded in Google Chrome dev tools while the second screenshot is generated with Google's https://pagespeed.web.dev (which would also use Lighthouse internally but with Chromium).
